### ISSUE TITLE
No seed option, so two identical GRPO commands cannot produce identical results

### ISSUE THREAD

**Issue by CodersAcademy006** [ellanorai/ThinkRL#83]
There is no `--seed` option on any CLI, and no call to `torch.manual_seed`, `random.seed` or `numpy.random.seed` on the training path. Generation samples with `do_sample=True` and `RLHFDataLoader` is constructed with `shuffle=True` at `grpo_trainer.py:132`, so every run differs.

Running the same command twice against an identical fixture produces different weights:

```
run A : {'transformer.h.0.attn.c_attn.weight': 49.185844, ...}
run B : {'transformer.h.0.attn.c_attn.weight': 49.185959, ...}
```

The practical consequence is that no change to the training code can be validated by comparing two runs. While investigating whether `--grad-accum` had any effect, I could not distinguish "the flag does nothing" from "the difference is sampling noise" empirically, and had to fall back on reading the call signature. Any future pull request touching the trainer faces the same problem, as does anyone trying to reproduce a benchmark number.

**Suggested request**

Add a `--seed` option that seeds Python, NumPy and torch, passes a generator to the dataloader, and is recorded in the run configuration. This is a precondition for reviewing or reproducing anything in the training path, so it may be worth doing before the algorithmic work.

I am happy to open a PR for this.
